gadget/validate.go

```diff
@@ -216,11 +216,12 @@ func validateVolumeContentsPresence(gadgetSnapRootDir string, vol *LaidOutVolume
 			continue
 		}
 		for _, c := range s.Content {
+			// TODO: detect and skip Content with "$kernel:" style refs
```

Comment:
what's the plan here?  to calll ResolveResouceContent optionally before Validate or not? or to pass kernel dir optionally to it and apply inside Validate?

Reply:
Yes, I think that is the sensible thing to do. We currently use "gadget.Validate()" in two places:
- "snap pack" will call it, here we don't know hat kernel will be used with it so we can't do more validation
- DeviceManager.doSetupRunSystem() will call it, here we do know the kernel and can validate that they work together.